## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 428733`
Commit: `f46dda8630bc55a1556c8cd2557fc1b221446bb5`

---
### `x86_64-linux`
<details>
  <summary>:x: 41 packages failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.mcp</li>
    <li>home-assistant-component-tests.mcp_server</li>
    <li>litellm</li>
    <li>litellm.dist</li>
    <li>mcp-proxy</li>
    <li>mcp-proxy.dist</li>
    <li>oterm</li>
    <li>oterm.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.smolagents</li>
    <li>python312Packages.smolagents.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python313Packages.django-mcp-server</li>
    <li>python313Packages.django-mcp-server.dist</li>
    <li>python313Packages.fastapi-mcp</li>
    <li>python313Packages.fastapi-mcp.dist</li>
    <li>python313Packages.fastmcp</li>
    <li>python313Packages.fastmcp.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.mcp</li>
    <li>python313Packages.mcp.dist</li>
    <li>python313Packages.mcpadapt</li>
    <li>python313Packages.mcpadapt.dist</li>
    <li>python313Packages.openai-agents</li>
    <li>python313Packages.openai-agents.dist</li>
    <li>python313Packages.private-gpt</li>
    <li>python313Packages.private-gpt.dist</li>
    <li>python313Packages.smolagents</li>
    <li>python313Packages.smolagents.dist</li>
    <li>python313Packages.txtai</li>
    <li>python313Packages.txtai.dist</li>
    <li>rembg</li>
    <li>rembg.dist</li>
    <li>vectorcode</li>
    <li>vectorcode.dist</li>
    <li>vimPlugins.vectorcode-nvim</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>python312Packages.django-mcp-server</li>
    <li>python312Packages.django-mcp-server.dist</li>
    <li>python312Packages.fastapi-mcp</li>
    <li>python312Packages.fastapi-mcp.dist</li>
    <li>python312Packages.fastmcp</li>
    <li>python312Packages.fastmcp.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.mcp</li>
    <li>python312Packages.mcp.dist</li>
    <li>python312Packages.mcpadapt</li>
    <li>python312Packages.mcpadapt.dist</li>
    <li>python312Packages.openai-agents</li>
    <li>python312Packages.openai-agents.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.smolagents</summary>
<pre>==================================== ERRORS ====================================
_____________________ ERROR collecting tests/test_types.py _____________________
ImportError while importing test module '/build/source/tests/test_types.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/2i76abyx06asxr2k3rgsp8zx647qj38h-python3-3.12.11/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_types.py:21: in <module>
    from transformers.testing_utils import (
E   ImportError: cannot import name 'require_soundfile' from 'transformers.testing_utils' (/nix/store/1qbmlivmw3zkgscmahsch9q6amnlvhz2-python3.12-transformers-4.54.0/lib/python3.12/site-packages/transformers/testing_utils.py)
=============================== warnings summary ===============================
tests/test_all_docs.py:107
  /build/source/tests/test_all_docs.py:107: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.timeout(100)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR tests/test_types.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
============ 40 deselected, 1 warning, 1 error in 60.21s (0:01:00) =============</pre>
</details>
<details>
<summary>python313Packages.mcp</summary>
<pre>                        traceback.format_exception(
                            cm.unraisable.exc_type,
                            cm.unraisable.exc_value,
                            cm.unraisable.exc_traceback,
                        )
                    )
>                   warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
E                   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO [closed]>
E                   
E                   Traceback (most recent call last):
E                     File "/nix/store/wa8b4x0vsvxz1dvgwnm9gdwhvba7ka70-python3.13-pydantic-2.11.7/lib/python3.13/site-packages/pydantic/json_schema.py", line 319, in build_schema_type_to_method
E                       mapping[key] = getattr(self, method_name)
E                                      ~~~~~~~^^^^^^^^^^^^^^^^^^^
E                   ResourceWarning: unclosed file <_io.FileIO name=18 mode='rb' closefd=True>

/nix/store/irg9b8cr3rxqkmgvalddzh5ps6lxzva3-python3.13-pytest-8.3.5/lib/python3.13/site-packages/_pytest/unraisableexception.py:85: PytestUnraisableExceptionWarning
=========================== short test summary info ============================
FAILED tests/client/test_stdio.py::TestChildProcessCleanup::test_early_parent_exit - AssertionError: Child should be writing
FAILED tests/issues/test_100_tool_listing.py::test_list_tools_returns_all_tools - pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO ...
============= 2 failed, 514 passed, 2 skipped, 1 xfailed in 56.06s =============</pre>
</details>
</details>

--- a/pkgs/development/python-modules/mcp/default.nix
+++ b/pkgs/development/python-modules/mcp/default.nix
@@ -11,6 +11,7 @@
   anyio,
   httpx,
   httpx-sse,
+  jsonschema,
   pydantic,
   pydantic-settings,
   python-multipart,
@@ -28,6 +29,7 @@
   websockets,
 
   # tests
+  dirty-equals,
   inline-snapshot,
   pytest-asyncio,
   pytest-examples,
@@ -38,20 +40,31 @@
 
 buildPythonPackage rec {
   pname = "mcp";
-  version = "1.9.4";
+  version = "1.12.2";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "modelcontextprotocol";
     repo = "python-sdk";
     tag = "v${version}";
-    hash = "sha256-VXbu/wHbXGS+cISJVUgCVEpTmZc0VfckNRoMj3GDi/A=";
+    hash = "sha256-K3S+2Z4yuo8eAOo8gDhrI8OOfV6ADH4dAb1h8PqYntc=";
   };
 
   postPatch = ''
     substituteInPlace pyproject.toml \
       --replace-fail ', "uv-dynamic-versioning"' "" \
       --replace-fail 'dynamic = ["version"]' 'version = "${version}"'
+  ''
+  + lib.optionalString stdenv.buildPlatform.isDarwin ''
+    # time.sleep(0.1) feels a bit optimistic and it has been flaky whilst
+    # testing this on macOS under load.
+    substituteInPlace \
+      "tests/client/test_stdio.py" \
+      "tests/server/fastmcp/test_integration.py" \
+      "tests/shared/test_ws.py" \
+      "tests/shared/test_sse.py" \
+      "tests/shared/test_streamable_http.py" \
+      --replace-fail "time.sleep(0.1)" "time.sleep(1)"
   '';
 
   build-system = [ hatchling ];
@@ -64,6 +77,7 @@ buildPythonPackage rec {
     anyio
     httpx
     httpx-sse
+    jsonschema
     pydantic
     pydantic-settings
     python-multipart
@@ -88,6 +102,7 @@ buildPythonPackage rec {
   pythonImportsCheck = [ "mcp" ];
 
   nativeCheckInputs = [
+    dirty-equals
     inline-snapshot
     pytest-asyncio
     pytest-examples
@@ -97,29 +112,45 @@ buildPythonPackage rec {
   ]
   ++ lib.flatten (lib.attrValues optional-dependencies);
 
-  pytestFlags = [
-    "-Wignore::pydantic.warnings.PydanticDeprecatedSince211"
-  ];
-
   disabledTests = [
     # attempts to run the package manager uv
     "test_command_execution"
 
-    # performance-dependent test
+    # ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
+    "test_lifespan_cleanup_executed"
+
+    # AssertionError: Child process should be writing
+    "test_basic_child_process_cleanup"
+
+    # AssertionError: parent process should be writing
+    "test_nested_process_tree"
+
+    # AssertionError: Child should be writing
+    "test_early_parent_exit"
+
+    # pytest.PytestUnraisableExceptionWarning: Exception ignored in: <_io.FileIO ...
+    "test_list_tools_returns_all_tools"
+
+    # AssertionError: Server startup marker not created
+    "test_stdin_close_triggers_cleanup"
+
+    # pytest.PytestUnraisableExceptionWarning: Exception ignored in: <function St..
+    "test_resource_template_client_interaction"
+
+    # Flaky: https://github.com/modelcontextprotocol/python-sdk/pull/1171
+    "test_notification_validation_error"
+
+    # Flaky: httpx.ConnectError: All connection attempts failed
+    "test_sse_security_"
+    "test_streamable_http_"
+
+    # This just feels a bit optimistic...
+    #     	assert duration < 3 * _sleep_time_seconds
+    # AssertionError: assert 0.0733884589999434 < (3 * 0.01)
     "test_messages_are_executed_concurrently"
 
     # ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
-    "test_client_session_version_negotiation_failure"
-
-    # AttributeError: 'coroutine' object has no attribute 'client_metadata'
-    "TestOAuthClientProvider"
-
-    # inline_snapshot._exceptions.UsageError: snapshot value should not change. Use Is(...) for dynamic snapshot parts
-    "test_build_metadata"
-  ]
-  ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    # Flaky: ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
-    "test_notification_validation_error"
+    "test_tool_progress"
   ];
 
   __darwinAllowLocalNetworking = true;

--- a/pkgs/development/python-modules/mcp/default.nix
+++ b/pkgs/development/python-modules/mcp/default.nix
@@ -160,6 +160,9 @@ buildPythonPackage rec {
     description = "Official Python SDK for Model Context Protocol servers and clients";
     homepage = "https://github.com/modelcontextprotocol/python-sdk";
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ josh ];
+    maintainers = with lib.maintainers; [
+      bryanhonof
+      josh
+    ];
   };
 }

--- a/pkgs/development/python-modules/openai-agents/default.nix
+++ b/pkgs/development/python-modules/openai-agents/default.nix
@@ -1,6 +1,5 @@
 {
   lib,
-  nix-update-script,
   buildPythonPackage,
   fetchPypi,
   hatchling,
@@ -15,19 +14,14 @@
 
 buildPythonPackage rec {
   pname = "openai-agents";
-  version = "0.0.13";
+  version = "0.2.3";
   pyproject = true;
 
   src = fetchPypi {
     inherit version;
     pname = "openai_agents";
-    hash = "sha256-a4AxXnXAa1MCxfKtui+eo4RflGFdrtRwa/uHF0D1YaU=";
+    hash = "sha256-ldStGUxcDPGkADjLcB7ujs2q92mNh7sT48LFz/gMS00=";
   };
-
-  # OpenAI 1.76.0 seems to not build currently
-  postPatch = ''
-    substituteInPlace pyproject.toml --replace-fail "openai>=1.76.0" "openai"
-  '';
 
   build-system = [
     hatchling
@@ -46,8 +40,6 @@ buildPythonPackage rec {
   pythonImportsCheck = [
     "agents"
   ];
-
-  passthru.updateScript = nix-update-script { };
 
   meta = {
     changelog = "https://github.com/openai/openai-agents-python/releases/tag/${version}";

